### PR TITLE
Add grafana-agent-logging-secret in the chart to store logging write credentials

### PR DESCRIPTION
**Pull request by marieroque**
towards https://github.com/giantswarm/giantswarm/issues/28486

We need to access secret data in the chart in order to configure the river config.
This river config specifies how we connect to loki and sensitive data must to be encrypted.

### CODE REVIEW

File: helm/grafana-agent/templates/logging-credentials-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: {{ .Values.loggingSecret.secretName | default "grafana-agent-secret" }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "grafana-agent.labels" . | nindent 4 }}
+stringData:
+  logging-url: {{ .Values.loggingSecret.url }}
+  logging-tenant-id: {{ .Values.loggingSecret.tenantId }}
+  logging-username: {{ .Values.loggingSecret.username }}
+  logging-password: {{ .Values.loggingSecret.password }}
```
Comment (QuantumEnigmaa) [inline helm/grafana-agent/templates/logging-credentials-secret.yaml:13]: Maybe you could add those fields in the `loggingSecret` section in the values with a comment to explain what their values should be

**Review by QuantumEnigmaa — APPROVED** [review]
LGTM

File: helm/grafana-agent/templates/logging-credentials-secret.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{{- if .Values.loggingSecret.enabled }}
```
Comment (QuentinBisson) [inline helm/grafana-agent/templates/logging-credentials-secret.yaml:1]: I like the idea of an extra secret but do not like the custom secret parameters. I would have liked to see a more generic approach that could have been proposed upstream as well like https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus/extrasecret.yaml for the secret and this to read the secret data https://grafana.com/docs/agent/latest/flow/reference/components/remote.kubernetes.secret/#example ?

Comment (marieroque) [reply helm/grafana-agent/templates/logging-credentials-secret.yaml:1]: I like your idea :-)

**Pull request merged**